app/src/hid.c

```diff
@@ -164,6 +170,18 @@ int zmk_hid_keyboard_release(zmk_key_t code) {
     return 0;
 };
 
+bool zmk_hid_keyboard_is_pressed(zmk_key_t code) {
+    if (code >= HID_USAGE_KEY_KEYBOARD_LEFTCONTROL && code <= HID_USAGE_KEY_KEYBOARD_RIGHT_GUI) {
+        return zmk_hid_mod_is_pressed(code - HID_USAGE_KEY_KEYBOARD_LEFTCONTROL);
+    }
+    for (int idx = 0; idx < CONFIG_ZMK_HID_KEYBOARD_REPORT_SIZE; idx++) {
+        if (keyboard_report.body.keys[idx] == code) {
+            return true;
+        }
+    }
```

Comment:
This doesn't handle both the NKRO and HKRO conditions. This should probably delegate to another function, e.g. `check_keyboard_usage(code)` that has different implementations for each Kconfig option, following other examples like `deselect_keyboard_usage`.

Reply:
I believe I have now addressed this. I added a test that uses NKRO as a check, and it all seems to be working. Let me know if you have other concerns with this. Thanks for the review!